hoggar/services/addSubscription.ts

```diff
@@ -0,0 +1,23 @@
+import { PrismaClient } from "@prisma/client";
+import { PrismaClientKnownRequestError } from "@prisma/client/runtime";
+
+const prisma = new PrismaClient();
+
+export async function addSubscription(email: string) {
+  const data = {
+    email: email,
+  };
+  try {
+    const subscriber = await prisma.subscription.create({
+      data: data,
+    });
+    return subscriber;
+  } catch (error) {
+    if (error instanceof PrismaClientKnownRequestError) {
+      if (error.code === "P2002") {
+        return "already subscribed !!!";
+      }

```

Comment:
should return error code not : message for human
```suggestion
      if (error.code === "P2002") {
        throw new Error("duplicate_email");
      }

```